blocksub/blocksub.go

```diff
@@ -201,6 +201,8 @@ func (s *BlockSub) _pollNow() error {
 // Also blocks if another instance is currently connecting.
 func (s *BlockSub) startWebsocket(retryForever bool) error {
 	if isAlreadyConnecting := s.wsIsConnecting.Swap(true); isAlreadyConnecting {
+		s.wsConnectingCond.L.Lock()
+		defer s.wsConnectingCond.L.Unlock()
```

Comment:
could you remove the defer and just move it one line down?

Reply:
done